### DISCUSSION TITLE
vsg::Inherit::create doesn't use perfect forwarding

### DISCUSSION THREAD

**Discussion by AnyOldName3** [vsg-dev/VulkanSceneGraph#1098] [Ideas]
The methods of `vsg::Inherit`, such as its constructor and `create` take forwarding references as variadic templated parameters, and then don't use `std::forward` to actually forward them to the other class. That means that the forwarding references decay to lvalue references, and any functions expecting forwarding/rvalue references won't get them. That in turn means that there'll be redundant copies that could have been cheaper moves instead.

The simple fix would be to change
```cpp
        template<typename... Args>
        static ref_ptr<Subclass> create(Args&&... args)
        {
            return ref_ptr<Subclass>(new Subclass(args...));
        }
```
to
```cpp
        template<typename... Args>
        static ref_ptr<Subclass> create(Args&&... args)
        {
            return ref_ptr<Subclass>(new Subclass(std::forward<Args>(args)...));
        }
```
and do the same for similar methods.

Was this just an oversight, or was there some intentional reason to disable forwarding?

**Comment by robertosfield** [comment]
An oversight, I assumed that it'd be passed on without changing type.

Try changing it, if everything builds cleanly we can merge this improvements.
